modules/utxo-lib/src/transaction.js

```diff
@@ -564,6 +564,41 @@ Transaction.prototype.hashForSignature = function (inIndex, prevOutScript, hashT
   return bcrypto.hash256(buffer)
 }
 
+/**
+ * Calculate the hash to verify the signature against
+ * @param inIndex
+ * @param prevoutScript
+ * @param value - The previous output's amount
+ * @param hashType
+ * @param isSegwit
+ * @returns {*}
+ */
+Transaction.prototype.hashForSignatureByNetwork = function (
+  inIndex,
+  prevoutScript,
+  value,
+  hashType,
+  isSegwit,
+) {
+  switch (coins.getMainnet(this.network)) {
+    case networks.bitcoingold:
+      return this.hashForGoldSignature(inIndex, prevoutScript, value, hashType, isSegwit)
+    case networks.bitcoincash:
+    case networks.bitcoinsv:
+      return this.hashForCashSignature(inIndex, prevoutScript, value, hashType)
+    case networks.zcash:
+      return this.hashForZcashSignature(inIndex, prevoutScript, value, hashType)
+    default:
+      if (isSegwit) {
+        return this.hashForWitnessV0(inIndex, prevoutScript, value, hashType)
+      } else {
+        return this.hashForSignature(inIndex, prevoutScript, hashType)
+      }
+  }
+
+  throw new Error('invalid state') // eslint-disable-line
```

Comment:
This looks like unreachable code, I see you're disabling the eslint warning but why have it at all? Nice to have this logic in its own reusable method though.

Reply:
Also, should this be within the `bitgo` subfolder since it appears to be bitgo specific logic? This file could be typescript too.

Reply:
While it is bitgo-specific, it would still be a overridden class method when the lib becomes a wrapper.

The `src/bitgo` should be for higher-level wrappers. I want to avoid dependencies from this code back.

Reply:
Re: unreachable code - this is just a habit. If the code changes and we don't hit a `return` we are still safe.